src/artery/envmod/CarIdentityRegistrant.cc

```diff
@@ -0,0 +1,41 @@
+#include "artery/envmod/CarIdentityRegistrant.h"
+#include "artery/utility/IdentityRegistry.h"
+#include "artery/traci/MobilityBase.h"
+
+
+namespace artery
+{
+
+Define_Module(CarIdentityRegistrant)
+
+void CarIdentityRegistrant::initialize()
+{
+	initializeIdentity();
+	registerIdentity();
+}
+
+void CarIdentityRegistrant::initializeIdentity()
+{
+	auto parent = this->getParentModule();
+	auto mobility = dynamic_cast<MobilityBase*>(parent->getSubmodule("mobility", -1));
+	auto controller = mobility->getVehicleController();
+
+	auto traciId = controller->getVehicleId();
+	auto stationId = rand();
```

Comment:
rand() is also used in [VehicleDataProvider](https://github.com/riebl/artery/blob/master/src/artery/application/VehicleDataProvider.cc#L66)
While the stationId is not critical for simulation reproducibility, should I use omnet++'s RNG here instead for good measure?

Reply:
Since CarIdentityRegistrant is an OMNeT++ module I suggest to go for OMNeT++'s RNG. We may want to change the behaviour by using VehicleDataProvider's non-default constructor to initialize its station ID from VehicleMiddleware.